### DISCUSSION TITLE
Wildcard DNS update does work on Cloudflare

### DISCUSSION THREAD

**Discussion by Yericus** [qdm12/ddns-updater#366] [General]
Hi,

Just wanted to add here that wildcard DNS updating does seem to work for cloudflare contrary to the cloudflare ddns updater documentation posted here :
[https://github.com/qdm12/ddns-updater/blob/master/docs/cloudflare.md](https://github.com/qdm12/ddns-updater/blob/master/docs/cloudflare.md)

I set up 2 entries for myexample.com in ddns-updater json config file.
One has domain myexample.com
One has domain *.myexample.com

Wildcard A entry has been created in Cloudflare DNS before.
ddns-updater shows success in updating the wildcard entry.

On another domain I am using where I had not added a wildcard A record already, ddns-updater showed failure, then success when said record was created.

Cloudflare has opened wildcard for free users in april/may 2022 which may explain why this is working now.

Might be useful for other users.

**Comment by qdm12** [comment]
Nice thanks for letting me know. So `ddns-updater` works with `*` and `@` already, does it work with a custom subdomain (i.e. `host.example.com`)? I also guess I should modify the create the record if it's not there 🤔

**Comment by Yericus** [comment]
```json
{
  "provider": "cloudflare",
  "zone_identifier": "xxx",
  "domain": "example.com",
  "host": "@",
  "ttl": 600,
  "token": "xxx",
  "ip_version": "ipv4"
}, {
  "provider": "cloudflare",
  "zone_identifier": "xxx",
  "domain": "*.example.com",
  "host": "@",
  "ttl": 600,
  "token": "xxx",
  "ip_version": "ipv4"
}
```

This is how I set up the json for updating both base domain and wildcard.
I guess updating a subdomain would work the same way, but would have to be created in the cloudflare DNS section first like the wildcard record needed too.

Sorry I did not test this myself though as wildcard is now catching everything and I am now just glad I got it to work somehow with Nginx Proxy Manager and Authelia instances for multiple domains.

I have no idea how host is working and that if creating records with the API is possible !

**Comment by qdm12** [comment]
Hi there, sorry for coming back 1+ year later, can you try with the latest image (dated february 2024 or later), the host as `*` and the domain as `example.com`?
I've been (finally) over the documentation to create a record at https://developers.cloudflare.com/api/operations/dns-records-for-a-zone-create-dns-record and it seems to match the existing code.
If it fails to create the record, can you share what the error is from the container logs?

Thanks!!

**Comment by Yericus** [comment]
I randomly had to put my head back in my docker setup as I realized my ip was not updated anymore.
I can't remember if I set something else on my pi hole machine or what not.

But I can tell you what previously worked for updating main domain name AND the wildcard does not work anymore.
The docker container is stuck in a shutdown loop and does not accept the '*' in the domain field as it was configured before.
I am a very sad panda.
```
2024-04-28 21:51:02 2024-04-28T14:51:02Z INFO reading JSON config from file /updater/data/config.json
2024-04-28 21:51:02 2024-04-28T14:51:02Z ERROR validating domain: domain has invalid character '*' at offset 0
2024-04-28 21:51:02 2024-04-28T14:51:02Z INFO Shutdown successful
```

```{
  "provider": "cloudflare",
  "zone_identifier": "xxx",
  "domain": "example.com",
  "host": "@",
  "ttl": 600,
  "token": "xxx",
  "ip_version": "ipv4"
}, {
  "provider": "cloudflare",
  "zone_identifier": "xxx",
  "domain": "*.example.com",
  "host": "@",
  "ttl": 600,
  "token": "xxx",
  "ip_version": "ipv4"
}



Using solution at https://github.com/qdm12/ddns-updater/issues/243#issuecomment-1584486445 to update wildcard works or so it seems.
Though for some reasons, my ips are all borked up and do not reflect my real IP.
Ugh ...

**Reply by qdm12** [reply]
The host has to be * and the domain example.com
